### PR TITLE
Support for meteor test command

### PR DESCRIPTION
**Pull request by jsep**

### CODE REVIEW

**Comment by bcbrian** [comment]
I forked this and merged into my master. :) Working great for me so far.

**Comment by DominikGuzei** [comment]
Hey there! I tried to use this with `METEOR@1.3-rc.7` and 

```
practicalmeteor:mocha@2.1.1-rc.1
practicalmeteor:chai
```
but i get this error when running `spacejam test`:

```
I20160328-13:31:49.952(2)? MochaRunner.runServerTests: Starting server side tests with run id Eq5baL8oDyXdnTZFY
W20160328-13:31:49.956(2)? (STDERR) MochaRunner.runServerTests: failures: 0
Exception from Tracker recompute function:
AssertionError: expected undefined to be a function
    at http://localhost:4096/packages/practicalmeteor_chai.js?hash=8fb9a4aa7055fa02f8b35e05fb1f05c8787c8a46:900
    at an (http://localhost:4096/packages/practicalmeteor_chai.js?hash=8fb9a4aa7055fa02f8b35e05fb1f05c8787c8a46:1131)
    at assert (http://localhost:4096/packages/practicalmeteor_chai.js?hash=8fb9a4aa7055fa02f8b35e05fb1f05c8787c8a46:3981)
    at ClientServerReporter (http://localhost:4096/packages/practicalmeteor_mocha.js?hash=81215895bcf916ea0cbd4f49d1218662f8ea051e:463)
    at http://localhost:4096/packages/practicalmeteor_mocha.js?hash=81215895bcf916ea0cbd4f49d1218662f8ea051e:2508
    at http://localhost:4096/packages/practicalmeteor_mocha.js?hash=81215895bcf916ea0cbd4f49d1218662f8ea051e:6920
    at http://localhost:4096/packages/practicalmeteor_mocha.js?hash=81215895bcf916ea0cbd4f49d1218662f8ea051e:366
    at http://localhost:4096/packages/tracker.js?hash=b267c37effdf4af331d16fc647c0370c2eaff44d:341
    at http://localhost:4096/packages/tracker.js?hash=b267c37effdf4af331d16fc647c0370c2eaff44d:360
    at http://localhost:4096/packages/tracker.js?hash=b267c37effdf4af331d16fc647c0370c2eaff44d:499
    at onGlobalMessage (http://localhost:4096/packages/meteor.js?hash=ec96c6f2d27fb027b17df77301967ce893b45963:391)
spacejam: phantomjs exited with code: 0
spacejam: killing meteor
spacejam: meteor killed with signal: SIGTERM
spacejam: All tests have passed. Exiting.
```

**Comment by jsep** [comment]
@DominikGuzei Please try the following:

```
meteor add practicalmeteor:mocha-console-runner@0.2.0-rc.1
spacejam test --mocha

```

File: src/CLI.coffee [hunk 1]
```diff
@@ -1,98 +1,100 @@
...
+    @spacejam.on 'done', (code)=>
+      if Spacejam.DONE_MESSAGE[code]?
+        exitMsg = "spacejam: #{Spacejam.DONE_MESSAGE[code]}. Exiting."
+      else
+        exitMsg = "spacejam: Unknown error with exit code '#{code}'. Exiting."
+      log.error exitMsg
+      process.exit code
+
+    try
+      @spacejam.doTests(command, @options)
```
Comment (rbabayoff) [inline src/CLI.coffee]: Rename to runTests

File: src/Spacejam.coffee [hunk 2]
```diff
@@ -97,7 +98,7 @@ class Spacejam extends EventEmitter
       @killChildren(Spacejam.DONE.METEOR_ERROR) if not @options.watch
 
     try
-      @meteor.testPackages(@options)
+      @meteor.doTestCommand(command, @options)
```
Comment (rbabayoff) [inline src/Spacejam.coffee]: Rename to runTests

File: src/Meteor.coffee [hunk 3]
```diff
@@ -51,8 +102,8 @@ class Meteor extends EventEmitter
 
   # @options
   # @parseCommandLine
-  testPackages: (options = {})=>
-    log.debug "Meteor.testPackages()", arguments
+  doTestCommand: (command, options = {})=>
```
Comment (rbabayoff) [inline src/Meteor.coffee]: runTestCommand

**Comment by serut** [comment]
You removed the file src/phantomjs-test-in-console.coffee on your branch `refactor`. 
Maybe create a new folder containing phantomjs scripts ?

**Comment by awatson1978** [comment]
Would love to see this merged in and have another console logging option available.  Using ``practicalmeteor:mocha-console-runner`` and ``dispatch:mocha-phantomjs`` with the ``meteor test`` command has been unreliable so far.

**Comment by icereed** [comment]
Any news on this? Not being able to test properly on my CI is quite a pain.

**Comment by awatson1978** [comment]
SWEET

Thank you, thank you!

**Comment by serut** [comment]
As I expected, travis can't run `bin/spacejam`, it throws this error:
```
	npm install -g https://github.com/practicalmeteor/spacejam/tarball/master coveralls
	spacejam test            --loglevel debug --driver-package=practicalmeteor:mocha-console-runner --coverage
	module.js:340
	throw err;
	^
	Error: Cannot find module '../lib/log'
	at Function.Module._resolveFilename (module.js:338:15)
	at Function.Module._load (module.js:280:25)
	at Module.require (module.js:364:17)
	at require (module.js:380:17)
	at Object.<anonymous> (/home/travis/.nvm/v0.10.40/lib/node_modules/spacejam/bin/spacejam:3:1)
	at Module._compile (module.js:456:26)
	at Object.Module._extensions..js (module.js:474:10)
	at Module.load (module.js:356:32)
	at Function.Module._load (module.js:312:12)
	at Function.Module.runMain (module.js:497:10)
	The command "spacejam test --loglevel debug --driver-package=practicalmeteor:mocha-console-runner --coverage" exited with 8.
```
You can find the entire log here: https://travis-ci.org/serut/meteor-coverage-app-exemple/builds/134144135

Node try to require the file `../lib/log.js` and not the file `../lib/log.coffee`. This is exactly this use case: https://stackoverflow.com/questions/4768748/requireing-a-coffeescript-file-from-a-javascript-file-or-repl
Please integrate my pull request https://github.com/practicalmeteor/spacejam/pull/57 that fix this issue.

**Pull request merged**